api/config/v1alpha1/helpers.go

```diff
@@ -147,6 +154,10 @@ func (r *ResourceProvider) GetKubeResourceProvider() *KubernetesResourceProvider
 		r.Kubernetes.EnvoyDeployment.Resources = DefaultResourceRequirements()
 	}
 
+	if r.Kubernetes.EnvoyDeployment.Image == nil {
+		r.Kubernetes.EnvoyDeployment.Image = DefaultKubernetesDeploymentImage()
```

Comment:
@AliceProxy raised a good point today about adding a warning log somewhere when user uses custom image 
it will help with triage, we can skip support for such setups

Reply:
We can add it later I think.